### DISCUSSION TITLE
GPort usage example

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#691] [Q&A]
Are there any examples out here showcasing `GPort` usage? When would you want to use a `GPort` as opposed to a `GNode`? E.g. would it make sense when a node has multiple points of connections for edges?

Also, I am seeing there is not corresponding builder for ports like `GNodeBuilder`. Is that intentional?

**Comment by planger** [comment]
Ports are typically used if there are multiple slots on a node to which you can connect edges, as for instance in this example: https://www.eclipse.org/glsp/gallery/#portsandrouting
Ports are usually children of those nodes and are placed by the GModelFactory relative to the containing node according to some logic. Typically they can not be moved by the user. Unfortunately, I'm not aware of any open-source example of using ports. But they are used just like nodes (contained by a node) and act as edge source and target. I don't think there is anything special to consider.

There is a `GPortBuilder` in the java library, but you are right, it is missing in the node library. This is not intentional but just likely hasn't been required yet and thus haven't been added. :/ Please feel free to open a bug request or a PR to add it. Thanks!

**Comment by rsoika** [comment]
@gee4vee  I have written some short documentation about the usage of ports and anchors in our own project. Maybe this can be helpful 
https://github.com/imixs/open-bpmn/blob/open-bpmn-0.9.3/doc/SPROTTY_ANCHORS_AND_PORTS.md

**Reply by srcnuzn** [reply]
Hi @rsoika, your link seems to be outdated (404 - page not found).

**Reply by rsoika** [reply]
Yes sorry, we moved the page. I fixed the [link](https://github.com/imixs/open-bpmn/blob/open-bpmn-0.9.3/doc/SPROTTY_ANCHORS_AND_PORTS.md) now.
